specification/analysisservices/resource-manager/Microsoft.AnalysisServices/stable/2017-08-01/analysisservices.json

```diff
@@ -1342,31 +1342,7 @@
     "GatewayError": {
       "description": "Detail of gateway errors.",
       "type": "object",
-      "properties": {
-        "code": {
-          "type": "string",
-          "description": "Error code of list gateway."
-        },
-        "message": {
-          "type": "string",
-          "description": "Error message of list gateway."
-        },
-        "subCode": {
-          "description": "The error sub code",
-          "type": "integer",
-          "format": "int32"
-        },
-        "httpStatusCode": {
-          "description": "The http status code",
-          "type": "integer",
-          "format": "int32"
-        },
-        "timeStamp": {
-          "description": "the timestamp for the error.",
-          "type": "string",
-          "readOnly": true
-        }
-      }
+      "$ref": "#/definitions/ErrorObject"
```

Comment:
This does NOT conform to the [ARM exception standard schema](https://github.com/Azure/azure-rest-api-specs/blob/bfbbca417537b86565c1f0434f6b0e34d0601811/specification/common-types/resource-management/v1/types.json#L277). Adding ARM Reviewer.

Reply:
The ErrorDetail does not contain these three properties:        
        "subCode": {
          "description": "The error sub code",
          "type": "integer",
          "format": "int32"
        },
        "httpStatusCode": {
          "description": "The http status code",
          "type": "integer",
          "format": "int32"
        },
        "timeStamp": {
          "description": "the timestamp for the error.",
          "type": "string",
          "readOnly": true
        },

I mean yes since the two properties are pretty much the same. if we are good to add these three properties in ErrorDetail,  we can fix it to use ErrorDetail.

Reply:
@allenjzhang is correct. The ARM error contract is documented at https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md#error-response-content. Deviating from it is not allowed.

Reply:
@majastrz  @allenjzhang hmm. I have all the required parameters in the ErrorObject. The properties in ErrorObject used be live in ErrorResponse but I took them since they can be shared with other error responses.
May I please have your suggestion or recommendation to fix this. Can we include  
"subCode": {
"description": "The error sub code",
"type": "integer",
"format": "int32"
},
"httpStatusCode": {
"description": "The http status code",
"type": "integer",
"format": "int32"
},
"timeStamp": {
"description": "the timestamp for the error.",
"type": "string",
"readOnly": true
},

in [ErrorDetail](https://github.com/Azure/azure-rest-api-specs/blob/70db689a2e699ff62550f5d7f62c4709e3b7882c/specification/common-types/resource-management/v2/types.json#L313)

Reply:
As ARM reviewer indicated, the ARM's response code is standardized. The subCode can be embedded in additionalInfo. HttpStatusCode is really necessary as it would be captured/reported on the client side. I don't know the scenario if the timeStamp is necessary or leveraged by the user. If you do want to include it as server's timeStamp, please include in the additinalInfo dictionary.